python/src/passes.cc

```diff
@@ -64,6 +64,7 @@ void init_triton_passes_ttgpuir(py::module &&m) {
                      createTritonGPUReduceDataDuplication);
   ADD_PASS_WRAPPER_0("add_allocate_shared_memory",
                      createAllocateSharedMemoryPass);
+  ADD_PASS_WRAPPER_0("add_canonicalizer", mlir::createCanonicalizerPass);
```

Comment:
why do you need that

Reply:
Due to the logic at https://github.com/triton-lang/triton/blob/main/lib/Dialect/TritonGPU/Transforms/CombineTensorSelectAndIf.cpp#L55-L61, it breaks upon encountering the first matching `scf.if`, which makes it unable to handle cases where multiple scf.if operations share the same condition (the same as select).

I need `canonicalize` pass to fold there `scf.if`

```mlir
%select = arith.select %cond, %trueVal, %falseVal
%if1 = scf.if %cond -> (...) {
  ...
  scf.yield ...
} else {
  scf.yield ...
}
%if2 = scf.if %cond -> (...) {
  ...
  scf.yield ...
} else {
  scf.yield ...
}
```

Reply:
done~